### ISSUE TITLE
Question: How to write a non-editor client that consumes the static analysis results? 

### ISSUE THREAD

**Issue by Symbolk** [microsoft/vscode-languageserver-node#505]
I am new to LSP. Previously I want to analyze programs with static analysis techniques (not developing a plugin or extension or editor), but I do not want to write one parser for each language, that leads me to the LSP. I was originally excited, but after studying for one week, I still do not know how to develop a non-editor client to interact with the language servers, because all examples I found are about plugin or extension development.

In a nutshell, I just want to send several code files in Java to the language server like eclipse.jdt.ls, and get definitions/references for symbols in these code files. Could any one give me a hand about how to do this simply in Java or Javascript?

**Comment by rcjsuen** [comment]
@Symbolk I think your question is perhaps more suited for the [microsoft/language-server-protocol repository](https://github.com/microsoft/language-server-protocol/) as this repository is about hooking language servers to Visual Studio Code which is clearly not your intent.

**Comment by Symbolk** [comment]
> @Symbolk I think your question is perhaps more suited for the [microsoft/language-server-protocol repository](https://github.com/microsoft/language-server-protocol/) as this repository is about hooking language servers to Visual Studio Code which is clearly not your intent.

Ok, thanks.

**Comment by rcjsuen** [comment]
@Symbolk Alternatively, you may want to pose your question in the [eclipse/eclipse.jdt.ls](https://github.com/eclipse/eclipse.jdt.ls) as there may be things that you need to properly bootstrap your system before talking to the language server (such as classpaths and so on).

**Comment by rcjsuen** [comment]
@Symbolk Silly me, I see you have already posted there. :)

**Comment by dbaeumer** [comment]
@Symbolk you can write a node client talking to a server process using `vscode-languageserver-protocol` module. Although VS Code in its name it only depends on node. 

Please be aware that this still requires that you start the server process. It only supports talking to an already running server by setting up the read and write stream yourself.
